subprojects/core/src/main/java/org/gradle/initialization/option/GradleBuildOptions.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.initialization.option;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+/**
+ * Defines all Gradle build options.
+ *
+ * @since 4.2
+ */
+public class GradleBuildOptions {
+
+    public static final GradleBuildStringOption DAEMON_IDLE_TIMEOUT;
+    public static final GradleBuildStringOption DAEMON_HEALTH_CHECK_INTERVAL;
+    public static final GradleBuildStringOption DAEMON_BASE_DIR;
+    public static final GradleBuildStringOption JVM_ARGS;
+    public static final GradleBuildStringOption JAVA_HOME;
+    public static final GradleBuildBooleanOption DAEMON;
+    public static final GradleBuildStringOption DEBUG_MODE;
+    public static final GradleBuildStringOption CONFIGURE_ON_DEMAND;
+    public static final GradleBuildStringOption PARALLEL;
+    public static final GradleBuildStringOption MAX_WORKERS;
+    public static final GradleBuildBooleanOption BUILD_CACHE;
+    public static final GradleBuildStringOption LOG_LEVEL;
+
```

Comment:
Aside, it seems like we could unify the logic and create options (properties) for most of our CLI flags, like "org.gradle.projectCacheDir".

Reply:
I don't quite understand this comment. Can you elaborate? Do you mean that every command line option _needs_ to have a corresponding Gradle property?

Reply:
Only for those that make sense to make persistent.